torch/nn/attention/__init__.py

```diff
@@ -162,3 +169,73 @@ def _sdpa_kernel_variadic(*backends: SDPBackend):
 def _get_flash_version() -> str:
     """This returns the closest matching tag for the flash attention backend"""
     return "2.5.7"
+
+
+_FlashAttentionBackend = Literal["FA4"]
+_FLASH_BACKEND_FA4: _FlashAttentionBackend = "FA4"
+
+_FLASH_ATTENTION_BACKENDS: dict[str, Callable[..., None]] = {}
```

Comment:
Maybe register_fn type should be a private TypeAlias since it's used a few places.